core/src/main/java/org/elasticsearch/index/query/HasParentQueryBuilder.java

```diff
@@ -57,32 +55,18 @@
 
     private final QueryBuilder<?> query;
     private final String type;
-    private boolean score = DEFAULT_SCORE;
-    private InnerHitBuilder innerHit;
-
-    /**
-     * @param type  The parent type
-     * @param query The query that will be matched with parent documents
-     */
-    public HasParentQueryBuilder(String type, QueryBuilder<?> query) {
-        if (type == null) {
-            throw new IllegalArgumentException("[" + NAME + "] requires 'parent_type' field");
-        }
-        if (query == null) {
-            throw new IllegalArgumentException("[" + NAME + "] requires 'query' field");
-        }
-        this.type = type;
-        this.query = query;
-    }
+    private final boolean score;
+    private final InnerHitBuilder innerHit;
```

Comment:
QueryBuilders should always set the defaults since they can be constructed without ever passing through the parser (Java API)